test/unittests.py

```diff
@@ -250,6 +255,36 @@ def test_generate_stats(self):
                                 results, regressions, progresses, new_cases)
         assert os.path.exists(FILE_PATHS['REPORT_DIFF_TXT_FILE'])
 
+    def test_gap_set_uuid16_svc_data(self):
+        advData = {}
+        # Test invalid inputs
+        with pytest.raises(ValueError, match="Invalid UUID16 value"):
+            gap_set_uuid16_svc_data(advData, 0xFFFF1, None)
+        with pytest.raises(ValueError, match="Invalid UUID16 value"):
+            gap_set_uuid16_svc_data(advData, -1, None)
+        with pytest.raises(TypeError):
+            gap_set_uuid16_svc_data(advData, 0xAAAA, 4)
+
+        # Test valid input
+        try:
+            gap_set_uuid16_svc_data(advData, 0x0000, None)
+            gap_set_uuid16_svc_data(advData, 0xFFFF, None)
+            gap_set_uuid16_svc_data(advData, 0xAABB, struct.pack('<B', 1))
```

Comment:
Since you added the support to cast to int, these could possibly be expanded. 

Additionally, we may want to consider just removing the attempt to cast, as you can cast most things to ints, including Boolean, floats, bytearrays, etc. 

<img width="801" height="224" alt="Image" src="https://github.com/user-attachments/assets/1a948176-0c41-40f7-bf0e-82c35a3d999c" />

Not sure if we want to support providing `122.5` as a valid UUID :D